Common/mivisionx_utils.hpp

```diff
@@ -0,0 +1,95 @@
+// MIT License
+//
+// Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#ifndef MIVISIONX_UTILS_HPP
+#define MIVISIONX_UTILS_HPP
+
+#include <VX/vx.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+
+/// \brief Macro to check OpenVX status and exit on error
+#define ERROR_CHECK_STATUS(status)                                                              \
+    {                                                                                           \
+        vx_status status_ = (status);                                                           \
+        if(status_ != VX_SUCCESS)                                                               \
+        {                                                                                       \
+            printf("ERROR: failed with status = (%d) at " __FILE__ "#%d\n", status_, __LINE__); \
+            exit(1);                                                                            \
+        }                                                                                       \
+    }
```

Comment:
```suggestion
#define ERROR_CHECK_STATUS(status)                                                                    \
    {                                                                                                \
        if(vx_status status_ = (status); status_ != VX_SUCCESS)                                                               \
        {                                                                                       \
            std::printf("ERROR: failed with status = (%d) at " __FILE__ "#%d\n", status_, __LINE__); \
            std::exit(EXIT_FAILURE);                                                                            \
        }                                                                                            \
    }
```

And the other macros, too.